src/main/scala/lambdaone/kratia/registry/Registry.scala

```diff
@@ -0,0 +1,41 @@
+package lambdaone.kratia.registry
+
+import cats.Monad
+import lambdaone.toolbox.{EventStore, Store}
+import cats.implicits._
+
+trait Registry[F[_], A] {
+
+  def isMember(community: Community[A], member: Member[A]): F[Boolean]
+
+  def load(community: Community[A], member: Member[A]): F[Option[A]]
+
+  def loadAll(community: Community[A]): F[List[A]]
+
+  def register(community: Community[A], member: Member[A], data: A): F[()]
+}
+
+object Registry {
+
+  implicit def registryWithEventStore[F[_], A](implicit store: EventStore[F, RegistryEvent[A]], F: Monad[F]): Registry[F, A] =
+    new Registry[F, A] {
+
+      override def isMember(community: Community[A], member: Member[A]): F[Boolean] =
+        store.find {
+          case RegistryEvent.RegisterMember(community0, member0, _) => community == community0 && member == member0
```

Comment:
The moment we introduce RegistryEvent.UnregisterMember we'll have to rewrite those. Seeing an entity state as the result of folding of all its events is the heart of EventSourcing and we should start reflecting this in code asap.

Reply:
Yhea that is true! I'll implement it on this one